controllers/routing/routing_delete_resources.go

```diff
@@ -6,44 +6,82 @@ import (
 
 	"github.com/opendatahub-io/odh-platform/pkg/metadata"
 	"github.com/opendatahub-io/odh-platform/pkg/metadata/labels"
+	"github.com/opendatahub-io/odh-platform/pkg/spi"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	ctrl "sigs.k8s.io/controller-runtime"
+	"k8s.io/apimachinery/pkg/selection"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-// HandleResourceDeletion handles the removal of dependent resources when the target resource is being deleted.
-func (r *PlatformRoutingController) HandleResourceDeletion(ctx context.Context, sourceRes *unstructured.Unstructured) (ctrl.Result, error) {
-	exportModes, found := extractExportModes(sourceRes)
-	if !found {
+func (r *PlatformRoutingController) removeUnusedRoutingResources(ctx context.Context, target *unstructured.Unstructured) error {
+	exportModes := extractExportModes(target)
+	unusedRouteTypes := spi.UnusedRouteTypes(exportModes)
+
+	if len(unusedRouteTypes) == 0 {
+		// no unused route types to remove resources for
+		return nil
+	}
+
+	var gvks []schema.GroupVersionKind
+	for _, unusedRouteType := range unusedRouteTypes {
+		gvks = append(gvks, routingResourceGVKs(unusedRouteType)...)
+	}
+
+	return r.deleteOwnedResources(ctx, target, unusedRouteTypes, gvks)
+}
+
+func (r *PlatformRoutingController) handleResourceDeletion(ctx context.Context, sourceRes *unstructured.Unstructured) error {
+	exportModes := extractExportModes(sourceRes)
+	if len(exportModes) == 0 {
 		r.log.Info("No export modes found, skipping deletion logic", "sourceRes", sourceRes)
 
-		return ctrl.Result{}, nil
+		return nil
 	}
 
 	r.log.Info("Handling deletion of dependent resources", "sourceRes", sourceRes)
 
+	var gvks []schema.GroupVersionKind
 	for _, exportMode := range exportModes {
-		if err := r.deleteOwnedResources(ctx, sourceRes, routingResourceGVKs(exportMode)); err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to delete resources for export mode %s: %w", exportMode, err)
-		}
+		gvks = append(gvks, routingResourceGVKs(exportMode)...)
+	}
+
+	if err := r.deleteOwnedResources(ctx, sourceRes, exportModes, gvks); err != nil {
+		return fmt.Errorf("failed to delete resources: %w", err)
 	}
```

Comment:
I don't think calling `removeUnusedRoutingResources` here will work out of the box as it does inverse logic - rather than deleting unused resources we deleted the 'used' resources